gma/src/include/firebase/gma/ump/types.h

```diff
@@ -0,0 +1,209 @@
+  }
+  /// Get the age of consent tag. This indicates whether the user is tagged for
+  /// under age of consent.
+  bool tag_for_under_age_of_consent() const {
+    return tag_for_under_age_of_consent_;
+  }
+  /// Get whether the age of consent tag was previously set.
+  bool has_tag_for_under_age_of_consent() const {
+    return has_tag_for_under_age_of_consent_;
+  }
+
+  /// Set the debug settings.
+  void set_debug_settings(const ConsentDebugSettings& settings) {
+    debug_settings_ = settings;
+    has_debug_settings_ = true;
+  }
+  /// Set the debug settings without copying data, which could be useful if you
+  /// have a large list of debug device IDs.
+  void set_debug_settings(const ConsentDebugSettings&& settings) {
+    debug_settings_ = std::move(settings);
+    has_debug_settings_ = true;
+  }
+  /// Get the debug settings.
+  const ConsentDebugSettings& debug_settings() const { return debug_settings_; }
+  /// Get whether debug settings were set.
+  bool has_debug_settings() const { return has_debug_settings_; }
+
+ private:
+  ConsentDebugSettings debug_settings_;
+  bool tag_for_under_age_of_consent_;
+  bool has_debug_settings_;
+  bool has_tag_for_under_age_of_consent_;
+};
+
+/// This is a platform specific datatype that is required to show a consent form
+/// on screen.
+///
+/// The following defines the datatype on each platform:
+/// <ul>
+///   <li>Android: A `jobject` which references an Android Activity.</li>
+///   <li>iOS: An `id` which references an iOS UIViewController.</li>
+/// </ul>
+#if FIREBASE_PLATFORM_ANDROID
+/// An Android Activity from Java.
+typedef jobject FormParent;
+#elif FIREBASE_PLATFORM_IOS || FIREBASE_PLATFORM_TVOS
+/// A pointer to an iOS UIViewController.
+typedef id FormParent;
+#else
+/// A void pointer for stub classes.
+typedef void* FormParent;
+#endif  // FIREBASE_PLATFORM_ANDROID, FIREBASE_PLATFORM_IOS,
+        // FIREBASE_PLATFORM_TVOS
+
+/// Consent status values.
+enum ConsentStatus {
+  /// Unknown status, e.g. prior to calling Request, or if the request fails.
+  kConsentStatusUnknown = 0,
+  /// Consent is required, but not obtained
+  kConsentStatusRequired,
+  /// Consent is not required
+  kConsentStatusNotRequired,
+  /// Consent was required, and has been obtained
+  kConsentStatusObtained
+};
+
+/// Errors that can occur during a RequestConsentInfoUpdate operation.
+enum ConsentRequestError {
+  /// The operation succeeded.
+  kConsentRequestSuccess = 0,
+  /// Invalid GMA App ID specified in AndroidManifest.xml or Info.plist.
+  kConsentRequestErrorInvalidAppId,
+  /// A network error occurred.
+  kConsentRequestErrorNetwork,
+  /// The tag for age of consent was not set. You must call
+  /// `ConsentRequestParameters::set_tag_for_under_age_of_consent()` before the
+  /// request.
+  kConsentRequestErrorTagForAgeOfConsentNotSet,
+  /// An internal error occurred.
+  kConsentRequestErrorInternal,
+  /// This error is undocumented.
```

Comment:
The iOS comment for this error is "A misconfiguration exists in the Funding Choices UI."
http://google3/googlemac/iPhone/GoogleAds/PersonalizedAdConsent/UserMessagingPlatform/source/public/UserMessagingPlatform/UMPError.h;l=12;rcl=532141888

We could probably just say "A misconfiguration exists in the UI."

Reply:
Done.